R/QC3e.R

```diff
@@ -0,0 +1,553 @@
+#' QC3e. Controle EC-gemeten & EC-berekend
+#'
+#' Vergelijking gemeten geleidbaarheid en berekende geleidbaarheid
+#' 
+#' Bereken de geleidbaarheid volgens de methode in Bijlage II van het protocol.
+#' 
+#' De signaleringswaarde voor monsters is delta-EC > 10 procent.
+#' Als de delta-EC boven de signaleringswaarde ligt, ken het
+#' concept QC oordeel twijfelachtig toe aan het monster.
+#'         
+#' @param d_metingen dataframe met metingen
+#' @param verbose of tekstuele output uit script gewenst is (T) of niet (F). Staat
+#' standaard op F.
+#'
+#' @return metingen bestand met verdachte locaties/monsters. 
+#'
+#' @export
+#'
+
+QC3e <- function(d_metingen, 
+                 ph_naam = "pH", 
+                 hco3_naam = "HCO3", 
+                 ec_naam = "GELDHD",
+                 celcius = 25,
+                 add_bicarbonate = F, 
+                 add_phosphate = F, 
+                 verbose = F) {
+
+  # Check datasets op kolommen en unieke informatie
+  testKolommenMetingen(d_metingen)
+  d <- d_metingen
+  
+  # aanpassen van opgegeven namen hco3 & ph & EC naar hco3 & hv & ecv. 
+  # Dat zijn de drie namen die gebruikt worden in de berekengeleidbaarheid functie
+  d <- d %>%
+    mutate(parameter = str_replace(parameter, ec_naam, "ecv")) %>%
+    mutate(parameter = str_replace(parameter, hco3_naam, "hco3v")) %>%
+    mutate(parameter = str_replace(parameter, ph_naam, "hv"))
+
+
+  # gegevens apart zetten om later qcid weer toe te voegen
+  id <- d %>%
+    dplyr::filter(parameter == "ecv") %>%
+    dplyr::select(qcid, monsterid, jaar, maand, dag, putcode, filter) 
+  
+  # selecteer relevante ionen om mee te nemen in ionenbalans
+  # h is pH en hv is pH-veld
+  an <- c("cl", "hco3", "hco3v", "no3", "so4", "co3", "po4", "ptot")
+  cat <- c("al", "ca", "fe", "k", "mg", "mn", "nh4", "na", "zn", "h", "hv")
+  
+  #zelfde parameter namen gebruiken als in de functies voor het berekenen van de geleidbaarheid
+  d$parameter <- tolower(d$parameter) 
```

Comment:
als HCO3 en HCO3_veld nu in het bestand opgenomen zijn en je kiest om als hco3_naam HCO3_veld te gebruiken worden beide in de berekening meegenomen. Moet alles worden aangepast zodat alleen hco3_naam wordt gebruikt of moet hco3_naam misschien worden hernoemd naar hco3_veld_naam? en als dat gebeurt misschien geen default waarde?